mix.exs

```diff
@@ -17,24 +21,49 @@ defmodule OpenTelemetry.MixProject do
         main: "OpenTelemetry"
         # logo: "path/to/logo.png",
         # extras: ["README.md"]
-      ]
+      ],
+      package: package(desc)
     ]
   end
 
-  defp version do
-    "VERSION"
+  defp version({:file, path}) do
+    path
     |> File.read!()
     |> String.trim()
   end
 
   # Run "mix help compile.app" to learn about applications.
-  def application do
-    []
-  end
+  def application, do: []
 
-  def deps() do
+  defp deps do
     [
       {:ex_doc, "~> 0.21", only: :dev, runtime: false}
     ]
   end
+
+  defp package(desc) do
+    links =
+      desc
+      |> Keyword.fetch!(:links)
+      |> Map.new(fn {key, value} -> {to_string(key), to_string(value)} end)
+    licenses = Keyword.fetch!(desc, :licenses)
+
+    [
+      build_tools: ~w(rebar3 mix),
+      licenses: licenses,
+      links: links
+    ]
+  end
+
+  defp load_config do
+    {:ok, config} = :file.consult('rebar.config')
+
+    config
+  end
+
+  defp load_app do
+    {:ok, [{:application, name, desc}]} = :file.consult('src/opentelemetry_api.app.src')
```

Comment:
This wouldn't work when used as a dependency if it was rebar3 running this script, but I take it `mix` changes directory to the root of each dep before building it? Did you verify this works if used as a git dep of another mix project?

Reply:
I'm pretty sure it would work, but if not then `File.cwd()` will always be the project root.

Reply:
What does "project root" mean in this context? If by project you mean the opentelemetry_api application, then good and there should be no need to add the path. 

But if "project root' means the project that has opentelemetry_api as a dependency then the current code won't work and not sure what you are to do with the cwd if that is the case.

Reply:
Sorry, that was unclear. I really just meant "it should be what you think it is", as in the root of *this*(opentelementry-api) project. I actually just confirmed this using some of my own repos.